.github/workflows/check.yml

```diff
@@ -1,53 +1,105 @@
-name: check
+name: Check
 
-on: [pull_request, push]
+on:
+  push:
+  pull_request:
+    branches:
+      - master
 
 jobs:
+  check_duplicate_runs:
+    name: Check for duplicate runs
+    continue-on-error: true
+    runs-on: ubuntu-latest
+    outputs:
+      should_skip: ${{ steps.skip_check.outputs.should_skip }}
+    steps:
+      - id: skip_check
+        uses: fkirc/skip-duplicate-actions@master
+        with:
+          concurrent_skipping: always
+          cancel_others: true
+          skip_after_successful_duplicate: true
+          paths_ignore: '["**/README.md", "**/CHANGELOG.md", "**/LICENSE.md"]'
+          do_not_skip: '["pull_request"]'
+
   mix_check:
     name: mix check (Elixir ${{matrix.elixir}}, OTP ${{matrix.otp}})
+    needs: check_duplicate_runs
+    if: ${{ needs.check_duplicate_runs.outputs.should_skip != 'true' }}
 
     strategy:
       matrix:
+        elixir:
+        - "1.7"
+        - "1.8"
+        - "1.9"
+        - "1.10"
+        otp:
+        - "20.3.8"
+        - "21.0"
+        - "22.0.2"
+        - "23.2.7"
+        exclude:
+        - elixir: "1.10"
+          otp: "20.3.8"
         include:
-          - elixir: '1.11.4'
-            otp: '23.2.7'
-          - elixir: '1.7.4'
-            otp: '20.3.8'
+        - elixir: "1.11.4"
+          otp: "23.2.7"
+          dialyzer: true
 
-    runs-on: ubuntu-20.04
+    runs-on: ubuntu-latest
```

Comment:
Using latest puts this setup at risk of starting to fail when next Ubuntu comes out and it's inconsistent with the rest of the config which uses hardcoded versions instead.

Reply:
OK